api/handler/v1/runtime.go

```diff
@@ -792,7 +792,7 @@ func (sv *RuntimeServiceServer) Replay(ctx context.Context, req *pb.ReplayReques
 		return nil, err
 	}
 
-	replayUUID, err := sv.jobSvc.Replay(replayWorkerRequest)
+	replayUUID, err := sv.jobSvc.Replay(ctx, replayWorkerRequest)
 	if err != nil {
 		return nil, status.Errorf(codes.Internal, "error while processing replay: %v", err)
```

Comment:
We should check for reply error type here and based on that return `codes.Internal` for general errors, `Unavailable` when the queue is full and can't push and `FailedPrecondition` when validation fails